staging/src/k8s.io/apiserver/plugin/pkg/audit/log/backend.go

```diff
@@ -17,77 +17,195 @@ limitations under the License.
 package log
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"strings"
+	"time"
 
+	"github.com/golang/glog"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
 	auditinternal "k8s.io/apiserver/pkg/apis/audit"
 	"k8s.io/apiserver/pkg/audit"
 )
 
 const (
+	// ModeDirect indicates that the backend should process events directly,
+	// Serialize event, then write it to logs. This mode spend more time.
+	ModeDirect = "direct"
+	// ModeBuffer indicates that the audit backend should buffer audit events
+	// internally, several consumers get event one by one and do write the
+	// real log. If the buffer is full, audit event will be dropped, so no
+	// block event producer.
```

Comment:
IMO part about several consumers is an implementation detail that is not important for the mode description

Also, `If the buffer is full, new audit events will be discarded.` seems like enough